src/app/plugins/animal.plugin.ts

```diff
@@ -0,0 +1,189 @@
+import { ChannelType, IContainer, IHttpResponse, IMessage, Maybe } from '../../common/types';
+import { MessageEmbed } from 'discord.js';
+import ms from 'ms';
+
+export default class AnimalPlugin extends Plugin {
+  public commandName: string = 'animal';
+  public name: string = 'Animal Plugin';
+  public description: string = 'Get wild with a fraction of gods creatures';
+  public usage: string =
+    'animal <Species (optional)> <subSpecies(Optional)> | animal list (species or <species> or subSpecies)';
+  public pluginAlias = ['wild', 'species'];
+  public permission: ChannelType = ChannelType.Public;
+  public pluginChannelName: string = Constants.Channels.Public.Pets;
+
+  private _API_URL: string = 'https://api.fetchit.dev/';
+
+  private _validAnimals: Set<string> = new Set([]); // All animals and species
+  private _species: string[] = []; // All species.
+  private _subspecies: ISubSpecies[] = []; // All subspecies housed by their species (not all species show here).
+
+  private _speciesEmbed: Maybe<MessageEmbed>;
+  private _subspeciesEmbed: Maybe<MessageEmbed>;
+
+  private _ANIMAL_UPD_THRESH: number = ms('1d');
+  private _LAST_UPD_TIME: number = 0;
+
+  constructor(public container: IContainer) {
+    super();
+    this._updateAnimalAPIData();
+  }
+
+  public async execute(message: IMessage, args?: string[]) {
+    this._updateAnimalAPIData();
+    const input = args?.length ? args : [this._getRandomAnimal()];
+
+    // Checks if they want to know what species/ subspecies are available.
+    if (input[0].startsWith('list')) {
+      if (!input[1] || input[1].startsWith('species')) {
+        message.reply(this._makeSpeciesEmbed());
+        return;
+      }
+
+      if (input[1].startsWith('subspecies')) {
+        message.reply(this._makeSubspeciesEmbed());
+        return;
+      }
+
+      await message.reply(this._makeSingleSubBreedEmbed(input[1]));
+      return;
+    }
+
+    // Preps the lookup data, and checks if it's valid.
+    const animalLookUp: string = input.join('_');
+    if (!this._validAnimals.has(animalLookUp)) {
+      message.reply(`\`${input}\`is a invalid species/ sub-species.`);
+      return;
+    }
+
+    let searchTerms: string = `name=${animalLookUp}`;
+    const speciesIndex = this._subspecies.findIndex((species) => species.species === animalLookUp);
+
+    // Checks if it's a sub-species, it retrieves its species.
+    if (!this._species.includes(animalLookUp)) {
+      const species = this._subspecies.find((species) => species.subSpecies.includes(animalLookUp))
+        ?.species;
+
+      searchTerms = `name=${species}&sub-species=${animalLookUp}`;
+    } else if (speciesIndex > -1) {
+      // Detects if the species has subspecies.  Then applies a random sub-species accordingly.
+      const subSpeciesList: string[] = this._subspecies[speciesIndex].subSpecies;
+      const randomSubspecies = subSpeciesList[Math.floor(Math.random() * subSpeciesList.length)];
```

Comment:
Yeah but one is pulling from a set and the other is pulling from an array, so it would just make function longer than it needs.

and the reason I have one for animal and not sub-species, is the line gets quite long for the random animal when it is inline, but sub-species is an okay length.

Open to changing it though, as right now it is not consistently handled.